python/ray/serve/config.py

```diff
@@ -254,3 +254,33 @@ def grpc_servicer_func_callable(self) -> List[Callable]:
                 raise ModuleNotFoundError(message) from e
 
         return callables
+
+
+@PublicAPI(stability="alpha")
+class EncodingType(str, Enum):
+    """Encoding type for the serve logs."""
+
+    TEXT = "TEXT"
+    JSON = "JSON"
+
+
+@PublicAPI(stability="alpha")
+class LoggingConfig(BaseModel):
+    """Logging config for configuring serve components logs.
+
+    Args:
+        encoding: Encoding type for the logs. Default to 'TEXT'. 'JSON' is
+            also supported to format all serve logs into json structure.
+        log_level: Log level for the serve logs. Default to logging.INFO.
+            You can set it to 'logging.DEBUG' to get more detailed logs.
+        logs_dir: Directory to store the logs. Default to None, which means
+            logs will be stored in the default directory
+            ("/tmp/ray/session_latest/logs/serve/...").
+        enable_access_log: Whether to write serve log to the file. Default to True.
+            When set to False, the serve logs are not written to files.
+    """
+
+    encoding: Optional[EncodingType] = EncodingType.TEXT
+    log_level: Optional[int] = logging.INFO
```

Comment:
Just learned those have numeric values XD Should we add an validation on the allowed values?

Reply:
Oh jeez it's not even an enum... https://github.com/python/cpython/blob/main/Lib/logging/__init__.py#L98

As this is written, in the config file users would need to look up the levels and convert them to an integer value... can we accept a `Union[None, str, int]` and convert passed strings to ints under the hood?